src/Prima/Pyxis/Modal.elm

```diff
@@ -0,0 +1,360 @@
+module Prima.Pyxis.Modal exposing
+    ( Config, config
+    , ModalSize, small, medium, large
+    , ModalHeader, defaultHeader, customHeader
+    , ModalFooter, emptyFooter, withButtonsFooter, customFooter
+    , view
+    )
+
+{-| Allows to create a Modal component using predefined Html syntax.
+
+
+# Modal Configuration
+
+@docs Config, config
+
+
+# Modal Size
+
+@docs ModalSize, small, medium, large
+
+
+# Modal Header
+
+@docs ModalHeader, defaultHeader, customHeader
+
+
+# Modal Footer
+
+@docs ModalFooter, emptyFooter, withButtonsFooter, customFooter
+
+
+# Rendering
+
+@docs view
+
+-}
+
+import Html exposing (..)
+import Html.Attributes exposing (class, classList)
+import Html.Events exposing (onClick)
+import Json.Decode
+import Prima.Pyxis.Button as Button
+
+
+{-| Represents the configuration of a Modal component.
+The Modal will trigger a "closed" Msg when the overlay is clicked.
+Is up to you to intercept and manage this Msg.
+-}
+type Config msg
+    = Config (Configuration msg)
+
+
+type alias Configuration msg =
+    { size : ModalSize
+    , header : ModalHeader msg
+    , content : List (Html msg)
+    , footer : ModalFooter msg
+    , closeTagger : msg
```

Comment:
invece di passare da fuori un messaggio che il componente deve lanciare per modificare il modello "contenitore" del modale, non sarebbe meglio che l'update ritorni lo stato aperto/chiuso come bool (il famigerato _terzo elemento della tupla_), da gestire quindi nell'update del padre?

Eliminerebbe la necessità di questa configurazione.

Reply:
Questo va fatto, il componente `Modal` è un refuso di Factotem fatto in velocità.